minitask2/akhmad/behaviour.py

```diff
@@ -0,0 +1,196 @@
+#!usr/bin/env python
+
+import rospy
+from tf.transformations import euler_from_quaternion
+
+from sensor_msgs.msg import LaserScan
+from geometry_msgs.msg import Twist
+from nav_msgs.msg import Odometry
+
+from math import radians, sqrt
+import random
+
+move_cmd = Twist()
+move_cmd.linear.x = 0.2
+
+left_turn_cmd = Twist()
+left_turn_cmd.linear.x = 0
+left_turn_cmd.angular.z = radians(30)
+
+right_turn_cmd = Twist()
+right_turn_cmd.linear.x = 0
+right_turn_cmd.angular.z = radians(-30)
+
+save_position = True
+last_pose_theta = None
+last_pose_x = None
+last_pose_y = None
+
+continue_right = False
+continue_left = False
+
+front = None
+right_front = None
+right_back = None
+right = None
+back = None
+left_back = None
+left_front = None
+left = None
+
+is_wall_found = False
+
+
+class Behaviour():
+    def __init__(self):
+        self.behaviour = input("Enter behaviour mode '0' for random walk '1' for right hand wall following: ")
```

Comment:
I believe these are meant to be happening at the same time. The robot should follow a wall if it finds one, or do a random walk if not.

Reply:
Yeah that's right. I separated these behaviours because my wall following implementation does not work properly